server/src/main/java/org/apache/iotdb/db/service/IoTDBShutdownHook.java

```diff
@@ -33,6 +35,11 @@ public void run() {
     CompactionTaskManager.getInstance().stop();
     // close rocksdb if possible to avoid lose data
     IoTDB.configManager.clear();
+
+    // == flush data to Tsfile and remove WAL log files
+    StorageEngine.getInstance().syncCloseAllProcessor();
+    WALManager.getInstance().deleteOutdatedWALFiles();
```

Comment:
whether could call IoTDB#stop() ?  other some service need clear together.

Reply:
Current codes changes have been tested. It can work well.

Now, I am not sure whether calling IoTDB#stop() will bring other issue. I will test it at once.

Reply:
> whether could call IoTDB#stop() ? other some service need clear together.

I just tested this. 
If we call IoTDB#stop() in IoTDBShutdownHook.java, it will be hanged for very long time. Because some IService can not exit soon.
So, if no special need, I prefer to keep current code change, which can get good closing speed and keep data safe.

Reply:
ok